src/reason-parser/reason_pprint_ast.ml

```diff
@@ -5629,8 +5629,13 @@ let printer = object(self:'self)
     | {pexp_desc = Pexp_constant constant} as x :: remaining ->
       let raw_literal, _ = extract_raw_literal x.pexp_attributes in
       self#formatChildren remaining (self#constant ?raw_literal constant :: processedRev)
-    | {pexp_desc = Pexp_construct ({txt = Lident "::"}, Some {pexp_desc = Pexp_tuple children} )} :: remaining ->
-      self#formatChildren (remaining @ children) processedRev
+    | {pexp_desc = Pexp_construct ({txt = Lident "::"}, Some {pexp_desc = Pexp_tuple children} )} as x :: remaining ->
+      let {jsxAttrs} = partitionAttributes x.pexp_attributes in
+      if jsxAttrs != [] then
```

Comment:
Another option is the pattern match on `jsxAttrs`.  Pattern matching is probably cheaper than physical equality checks. (Correct me if I'm wrong).

Reply:
Pattern matching should be safer to against future changes in compiler optimizations around equality.

Reply:
Hah, I actually had a pattern match at first but decided to go with this approach given it’s used everywhere else that does the same thing.